astropy/io/fits/header.py

```diff
@@ -105,7 +112,7 @@ def __init__(self, cards=[], txtfile=None):
             return
 
         if isinstance(cards, Header):
-            cards = cards.cards
+            cards = (copy.copy(card) for card in cards._cards)
```

Comment:
`Header` does not allow "mutable types" as key/value so there is no need to do more than a shallow copy. I'm always wary of `deepcopy` because it's really inefficient and sometimes does the wrong thing. Note that [`fits.Card`](https://github.com/astropy/astropy/blob/master/astropy/io/fits/card.py#L40) also contains a lot of regexes that shouldn't be deepcopied 😄